language/move-stdlib/tests/string_tests.move

```diff
@@ -75,4 +75,75 @@ module std::string_tests {
         string::insert(&mut s, 1, string::utf8(b"xy"));
         assert!(s == string::utf8(b"axybcd"), 22)
     }
+
+    #[test]
+    fun test_next_char_boundary() {
+        // using utf8 byte string ("\xf0\x9f\x99\x82") to represent a smiley emoji
+        let s = string::utf8(b"Smile \xf0\x9f\x99\x82 because it's move!");
+        let b = string::next_char_boundary(&s, 6);
+
+        assert!(b == 13, 22)
+    }
+
+    #[test]
+    #[expected_failure(abort_code = 2, location = std::string)]
+    fun test_next_char_boundary_invalid_index() {
+        let s = string::utf8(b"abcdefgh");
+        string::next_char_boundary(&s, 8);
+    }
+
+    #[test]
+    fun test_chars_count() {
+        // using utf8 byte string ("\xf0\x9f\x99\x82") to represent a smiley emoji
+        let s = string::utf8(b"\xf0\x9f\x99\x82 move?");
+        let b = string::chars_count(&s);
+
+        assert!(b == 7, 22)
+    }
+
+    #[test]
+    fun test_sub_string_char() {
+        // using utf8 byte string ("\xf0\x9f\x99\x82") to represent a smiley emoji
+        let s = string::utf8(b"\xf0\x9f\x99\x82 move?");
+
+        let sub1 = string::sub_string_char(&s, 2, 6);
+        assert!(sub1 == string::utf8(b"move"), 22);
+
+        let sub2 = string::sub_string_char(&s, 0, 10);
+        assert!(sub2 == s, 22)
+    }
+
+    #[test]
+    fun test_sub_string_char_empty() {
+        // using utf8 byte string ("\xf0\x9f\x99\x82") to represent a smiley emoji
+        let s = string::utf8(b"\xf0\x9f\x99\x82 move?");
+
+        let sub1 = string::sub_string_char(&s, 7, 7);
+        assert!(string::is_empty(&sub1), 22);
+
+        let sub2 = string::sub_string_char(&s, 7, 10);
+        assert!(string::is_empty(&sub2), 22);
+
+        let sub3 = string::sub_string_char(&s, 5, 3);
+        assert!(string::is_empty(&sub3), 22)
```

Comment:
Originally I thought the expected behavior for this should be that if you try to access an out-of-bounds character, an abort will occur, but this is not the case in Rust.
```
    let s = "Hello, wãrld!";
    let ss: String = s.chars().skip(7).take(99).collect();
```
works fine for example

Reply:
yeah, it'll work fine; aborting depends on the operation you're performing on the string.